core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/store/host-folder-field.store.ts

```diff
@@ -0,0 +1,120 @@
+        loadSites: rxMethod<string | null>(
+            pipe(
+                tap(() => patchState(store, { status: 'pending' })),
+                switchMap((path) => {
+                    return dotEditContentService
+                        .getSitesTreePath({ perPage: 7000, filter: '*' })
+                        .pipe(
+                            tapResponse({
+                                next: (sites) => patchState(store, { tree: sites }),
+                                error: () => patchState(store, { status: { error: 'fail' } }),
+                                finalize: () => patchState(store, { status: 'fulfilled' })
+                            }),
+                            map((sites) => ({
+                                path,
+                                sites
+                            }))
+                        );
+                }),
+                filter(({ path }) => !!path),
+                switchMap(({ path, sites }) => {
+                    const hasPaths = path.includes('/');
+                    if (!hasPaths) {
+                        const response: CustomTreeNode = {
+                            node: sites.find((item) => item.key === path),
+                            tree: null
+                        };
+
+                        return of(response);
+                    }
+
+                    return dotEditContentService.buildTreeByPaths(path);
+                }),
+                tap(({ node, tree }) => {
+                    const changes: Partial<HostFolderFiledState> = {};
+                    if (node) {
+                        changes.nodeSelected = node;
+                    }
+
+                    if (tree) {
+                        const currentTree = store.tree();
+                        const newTree = currentTree.map((item) => {
+                            if (item.key === tree.path) {
+                                return {
+                                    ...item,
+                                    children: [...tree.folders]
+                                };
+                            }
+
+                            return item;
+                        });
+                        changes.tree = newTree;
+                    }
+
+                    patchState(store, changes);
+                })
+            )
+        ),
+        loadChildren: rxMethod<TreeNodeSelectItem>(
+            pipe(
+                exhaustMap((event: TreeNodeSelectItem) => {
+                    const { node } = event;
+                    const { hostname, path } = node.data;
+
+                    return dotEditContentService.getFoldersTreeNode(hostname, path).pipe(
+                        tap((children) => {
+                            node.leaf = true;
+                            node.icon = 'pi pi-folder-open';
+                            node.children = children;
+                            node.expanded = true;
```

Comment:
can we use an object and do only one assign?

Reply:
@zJaaal done!